api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -226,6 +243,70 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
     return ResponseEntity.ok(previewQueryResponse);
   }
 
+  @Override
+  public ResponseEntity<EmptyResponse> exportToNotebook(String workspaceNamespace, String workspaceId, DataSetExportRequest dataSetExportRequest) {
+    workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
+    DataSetQueryList queryList = generateQuery(workspaceNamespace, workspaceId, dataSetExportRequest.getDataSetRequest()).getBody();
+    String queriesAsStrings = "import pandas\n\n" + queryList.getQueryList().stream()
+        .map(query -> convertQueryToString(query, dataSetExportRequest.getDataSetRequest().getName())).collect(Collectors.joining("\n\n"));
+    JSONObject notebookFile = new JSONObject()
+        .put("cells", new JSONArray()
+            .put(new JSONObject()
+                .put("cell_type", "code")
+                .put("metadata", new JSONObject())
+                .put("execution_count", JSONObject.NULL)
+                .put("outputs", new JSONArray())
+                .put("source", new JSONArray()
+                  .put(queriesAsStrings))))
+        .put("metadata", new JSONObject())
+        .put("nbformat", 4)
+        .put("nbformat_minor", 2);
+
+    WorkspaceResponse workspace = fireCloudService.getWorkspace(workspaceNamespace, workspaceId);
+
+    cloudStorageService.writeFile(workspace.getWorkspace().getBucketName(),
+        "notebooks/" + dataSetExportRequest.getNotebookName() + ".ipynb", notebookFile.toString().getBytes(StandardCharsets.UTF_8));
+
+
+    return ResponseEntity.ok(new EmptyResponse());
+  }
+
+  private String convertQueryToString(DataSetQuery query, String prefix) {
+    String namespace = prefix.toLowerCase().replaceAll(" ", "_") + "_" + query.getDomain().toString().toLowerCase() + "_";
+    String sqlSection = namespace + "sql = \"\"\"" + query.getQuery() + "\"\"\"";
```

Comment:
Question: Parameter name namespaceSql will be repeated for each domain so is it going to cause issue for duplicate variable name?

Reply:
It should be a different namespace each time, because the domain will be different for each query.